src/mount/mount-tool.c

```diff
@@ -1480,27 +1462,24 @@ static int list_devices(void) {
 
         (void) pager_open(arg_pager_flags);
 
-        fputs(ansi_underline(), stdout);
-        for (c = 0; c < _COLUMN_MAX; c++) {
-                if (c > 0)
-                        fputc(' ', stdout);
-
-                printf("%-*s", (int) column_width[c], titles[c]);
+        table = table_new("NODE", "PATH", "MODEL", "WWN", "TYPE", "LABEL", "UUID");
+        if (!table) {
+                log_oom();
+                goto finish;
         }
-        fputs(ansi_normal(), stdout);
-        fputc('\n', stdout);
 
-        for (i = 0; i < n; i++) {
+        for (i = 0; i < n; i++)
                 for (c = 0; c < _COLUMN_MAX; c++) {
-                        if (c > 0)
-                                fputc(' ', stdout);
-
-                        printf("%-*s", (int) column_width[c], strna(items[i].columns[c]));
+                        r = table_add_cell(table, NULL, TABLE_STRING, strna(items[i].columns[c]));
+                        if (r < 0) {
+                                log_error_errno(r, "Failed to add cell: %m");
+                                goto finish;
+                        }
                 }
-                fputc('\n', stdout);
-        }
 
-        r = 0;
+        r = table_print(table, NULL);
+        if (r < 0)
+                log_error_errno(r, "Failed to print table: %m");
 
 finish:
         for (i = 0; i < n; i++)
```

Comment:
so i think doing it this way is OK. But I think an even better approach would be to get rid of `items[]` and just add directly to the table. we don't need `items[]` only for sorting after all, but Table can already do that.

To be fully equivalent we'd have in introduce a new "path" cell type, which is like the "string" one, but uses path_compare() instead of strcmp() for ordering things when sorting.